### PR TITLE
Topic recognition - s4764408

### PR DESCRIPTION
**Pull request by pulpasaur**
This pull request proposes to integrate the solution for devising a multi-layer Graph Convolutional Network (GCN) model designed for semi-supervised multi-class node classification, utilizing the Facebook Large Page-Page Network dataset. The accuracy of this model is an average of 94% with a t-SNE visualization.

the files in the solutions for this pull request are as follows :

- dataset.py: Code for reprocessing and loading data

- modules.py: Source code of the component for the GCN 

- train.py: Source code for training, validating, testing and plotting the model

- predict.py: shows example use of the trained model, prints out any results

- README.MD: This project documentation

- Some images containing training results

Dependencies
- torch
- torch_geometric
- numpy
- sklearn

Joel Adhika Rian Forrel
s4764408

### CODE REVIEW

**Comment by gayanku** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Normal

Readme Overall: Ok, needs instructions on how to setup and run, especially the data. Provided code snippet screenshots are not relevant to usage and can be made redundant.

- Project Overview: Fair
- Model: No details.
- Data + Preprocessing: Good
- Training / Loss Curve(s): Good
- Result Demonstration: Good
- References: Good

Functionality:

- Multi-layer GCN: Fair, 3 layers. **uses external code** (torch_geometric.nn import GCNConv). hard coded hyper parameter.
- Semi supervised: Good
- Accuracy: Good
- TSNE or UMAP: Good
- Discussion: Good

Code:

- Consistent with Results: Yes
- File Structure: Good
- Commenting: Good
- Commit frequency: **All commits on Oct 20, 2023 (10 in total)**
- Commit messages: OK.

Other Comments:
Uses torch_geometric.nn import GCNConv module which is **not own GCN code**. 
No early stop.
No hyper parameter search.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Normal -5
## Commit Log
Meaningful commit messages minimal -1
Progressive commits used all in one day -1
## Documentation
ReadMe acceptable, no usage, no refs, problem description etc. -3
Model/technical explanation minimal -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good

**Comment by wangzhaomxy** [comment]
Please remove .gitignore file to eliminate the merge conflict.

**Comment by wangzhaomxy** [comment]
Feedback attempt and no feedback marks lost.
